### PR TITLE
Update sbt.bat to match bash sbt command line usage

### PR DESCRIPTION
**Pull request by er1c**
This started as an initial scrub of the `sbt.bat` but is shaping up to be a full port of the bash `sbt` to windows bat `sbt.bat`

* Implement (nearly all?) of the bash `sbt` `--help` arguments
* Windows now uses the same `integrationTest/test`

TODO:

- [ ] parse input `SBT_OPTS` (environment `SBT_OPTS`, `.sbtopts`, and `sbtconfig.txt`?) for arguments (e.g. `--no-color` and convert to appropriate `-Dsbt.log.noformat=true` options)
- [ ] Process memory arguments from SBT_OPTS (e.g. `-Xms2048M -Xmx2048M -Xss6M`)

### CODE REVIEW

**Comment by er1c** [comment]
Does `https://ci.appveyor.com/project/sbt/sbt-launcher-package` need to be removed from `README.md` and hook?  https://github.com/sbt/sbt-launcher-package/commit/261ddc3c3849cc85292897e55125e0939b2d6385

**Comment by eed3si9n** [comment]
Ideally we should be able to run https://github.com/sbt/sbt-launcher-package/blob/master/integration-test/src/test/scala/RunnerTest.scala on Windows.

**Comment by er1c** [comment]
> Does `https://ci.appveyor.com/project/sbt/sbt-launcher-package` need to be removed from `README.md` and hook? [261ddc3](https://github.com/sbt/sbt-launcher-package/commit/261ddc3c3849cc85292897e55125e0939b2d6385)

Does it currently not run?  Or is it failing in the CI?

**Comment by er1c** [comment]
> Ideally we should be able to run [/integration-test/src/test/scala/RunnerTest.scala@`master`](https://github.com/sbt/sbt-launcher-package/blob/master/integration-test/src/test/scala/RunnerTest.scala) [](/sbt/sbt-launcher-package/blob/HEAD@{2019-09-24T18:55:08Z}/integration-test/src/test/scala/RunnerTest.scala) on Windows.

Does it currently not run?  Or does it just fail?

**Comment by er1c** [comment]
I'll look into the CI failure, on windows I was getting:

```
C:\Work\sbt-launcher-package>sbt -Dsbt.build.version=1.3.0 universal:packageBin universal:stage integrationTest/test
[info] Loading global plugins from C:\Users\Eric Peters\.sbt\0.13\plugins
[info] Updating {file:/C:/Users/Eric%20Peters/.sbt/0.13/plugins/}global-plugins...
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by sbt.ivyint.ErrorMessageAuthenticator$ (file:/C:/Users/Eric%20Peters/.sbt/boot/scala-2.10.7/org.scala-sbt/sbt/0.13.17/ivy-0.13.17.jar) to field java.net.Authenticator.theAuthenticator
WARNING: Please consider reporting this to the maintainers of sbt.ivyint.ErrorMessageAuthenticator$
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
[info] Loading project definition from C:\Work\sbt-launcher-package\project
[info] Updating {file:/C:/Work/sbt-launcher-package/project/}sbt-launcher-package-build...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
[info] Compiling 1 Scala source to C:\Work\sbt-launcher-package\project\target\scala-2.10\sbt-0.13\classes...
[warn] Missing bintray credentials. Either create a credentials file with the bintrayChangeCredentials task, set the BINTRAY_USER and BINTRAY_PASS environment variables or pass bintray.user and bintray.pass properties to sbt.
[warn] Missing bintray credentials. Either create a credentials file with the bintrayChangeCredentials task, set the BINTRAY_USER and BINTRAY_PASS environment variables or pass bintray.user and bintray.pass properties to sbt.
[warn] Missing bintray credentials. Either create a credentials file with the bintrayChangeCredentials task, set the BINTRAY_USER and BINTRAY_PASS environment variables or pass bintray.user and bintray.pass properties to sbt.
[warn] Missing bintray credentials. Either create a credentials file with the bintrayChangeCredentials task, set the BINTRAY_USER and BINTRAY_PASS environment variables or pass bintray.user and bintray.pass properties to sbt.
[warn] Missing bintray credentials. Either create a credentials file with the bintrayChangeCredentials task, set the BINTRAY_USER and BINTRAY_PASS environment variables or pass bintray.user and bintray.pass properties to sbt.
[warn] Missing bintray credentials. Either create a credentials file with the bintrayChangeCredentials task, set the BINTRAY_USER and BINTRAY_PASS environment variables or pass bintray.user and bintray.pass properties to sbt.
[warn] Missing bintray credentials. Either create a credentials file with the bintrayChangeCredentials task, set the BINTRAY_USER and BINTRAY_PASS environment variables or pass bintray.user and bintray.pass properties to sbt.
[warn] Missing bintray credentials. Either create a credentials file with the bintrayChangeCredentials task, set the BINTRAY_USER and BINTRAY_PASS environment variables or pass bintray.user and bintray.pass properties to sbt.
java.lang.IllegalArgumentException: requirement failed: Revision cannot be empty.
        at scala.Predef$.require(Predef.scala:233)
        at sbt.StringUtilities$.nonEmpty(StringUtilities.scala:11)
        at sbt.impl.GroupArtifactID.$percent(DependencyBuilders.scala:49)
        at $6aff4b28fe4c52defe07$.sbtActual$lzycompute(build.sbt:28)
        at $6aff4b28fe4c52defe07$.sbtActual(build.sbt:28)
        at $6aff4b28fe4c52defe07$$anonfun$dist$9.apply(build.sbt:352)
        at $6aff4b28fe4c52defe07$$anonfun$dist$9.apply(build.sbt:352)
        at sbt.EvaluateSettings$$anonfun$sbt$EvaluateSettings$$constant$1.apply(INode.scala:163)
        at sbt.EvaluateSettings$$anonfun$sbt$EvaluateSettings$$constant$1.apply(INode.scala:163)
        at sbt.EvaluateSettings$MixedNode.evaluate0(INode.scala:175)
        at sbt.EvaluateSettings$INode.evaluate(INode.scala:135)
        at sbt.EvaluateSettings$$anonfun$sbt$EvaluateSettings$$submitEvaluate$1.apply$mcV$sp(INode.scala:69)
        at sbt.EvaluateSettings.sbt$EvaluateSettings$$run0(INode.scala:78)
        at sbt.EvaluateSettings$$anon$3.run(INode.scala:74)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
[error] java.lang.IllegalArgumentException: requirement failed: Revision cannot be empty.
[error] Use 'last' for the full log.
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore?

```

Does this ring a bell at all?

**Comment by eed3si9n** [comment]
@er1c It would fail.

**Comment by er1c** [comment]
> @er1c It would fail.

👍 I get it now I'll look into this, I agree it would be great!

File: integration-test/src/test/scala/RunnerTest.scala [hunk 1]
```diff
@@ -13,7 +13,8 @@ object SbtRunnerTest extends SimpleTestSuite with PowerAssertions {
   def sbtProcessWithOpts(arg: String, javaOpts: String, sbtOpts: String) =
     sbt.internal.Process(sbtScript.getAbsolutePath + " " + arg, new File("citest"),
       "JAVA_OPTS" -> javaOpts,
-      "SBT_OPTS" -> sbtOpts)
+      "SBT_OPTS" -> s"""$sbtOpts -Dsbt.global.base=C:\\Work\\.sbt\\ -Dsbt.ivy.home=C:\\Work\\.ivy2\\ -Divy.home=C:\\Work\\.ivy2\\ """
```
Comment (er1c) [inline integration-test/src/test/scala/RunnerTest.scala]: booh

File: src/universal/bin/sbt.bat [hunk 2]
```diff
@@ -172,103 +430,349 @@ if defined JVM_DEBUG_PORT (
...
+echo In the case of duplicated or conflicting options, the order above
+echo shows precedence: JAVA_OPTS lowest, command line options highest.
+echo.
+
+@endlocal
+exit /B 1
+
+:set_sbt_version
+rem print sbtVersion
+for /F "usebackq tokens=2" %%G in (`CALL "!_JAVACMD!" !JAVA_OPTS! !_SBT_OPTS! -cp "!SBT_HOME!\bin\sbt-launch.jar" xsbt.boot.Boot "sbtVersion" 2^>^&1`) do set "sbt_version=%%G"
```
Comment (er1c) [inline src/universal/bin/sbt.bat]: Looks like it's getting stuck here...I suspect something like:

```
[error] java.lang.ExceptionInInitializerError
[error] Use 'last' for the full log.
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore?
```

Is popping up.

**Pull request merged**